### DISCUSSION TITLE
Tips to fix AAI connection Issues

### DISCUSSION THREAD

**Discussion by Tuerkay** [BimmerGestalt/AAIdrive#268] [Show and tell]
Hi all,

i´ve posted this guide in the Bimmerfest Forum but it might be also useful here. i´ve tested AAI in many different cars with factory navigation and retrofitted ones. Cars from family and friend and also rental cars.
i´ve also talked with the Connected Drive support a few times about connection issues with the different BMW android apps and how to fix them. According to them, the BMW apps don´t work properly when updated through the app store. 
i always keep AAI up to date (always use the newest build) and also keep spotify updated. Since i follow the steps below, i never had connection issues.

1. Im some cars, BMW Apps in car only worked if telematic module had an network signal when HU booted. (If car was in an well isolated garage when HU booted, BMW Apps didn´t show up in Connected Drive menu. (seen on NBT and NBT Evo))
2. you should only have ONE bmw app installed on every time. e.g. if you have BMW Connected and myBMW installed on your phone at the same time. they´ll sometimes block each other and your smartphone won´t connect to car or connection will be unstable.
3. If you change or update/downgrade any BMW app. DON´T update directly. Go to app settings -> delete app data -> delete app -> reboot device -> install new BMW App.
4. Use the correct BMW App for your system. (my experience so far)

NBT: BMW Connected Classic 1.8 (BMW Connected 6.4 was very unreliable on getting a working connection) (wired)
NBT Evo id4: couldn´t find a car to test
NBT Evo id5/id6 (retrofitted): BMW Connected 6.4 (wireless/BT)
NBT Evo id5/id6 (factory <03/2018): BMW Connected 6.4 (wireless/BT)
NBT Evo id5/id6 (factory >=03/2018): BMW Connected 6.4 and myBMW (BMW Connected more stable and connects faster) (wireless/BT)

After updating to current BMW Connected 6.5, i wasn´t able to connect to any BMW properly after.

Hope this will also help others.

Edit2: myBMW: if your doing mostly long distance trips, you should consider using BMW connected even with newer cars. I mainly tested myBMW on short distance trips (20 - 30 min). Test driving a car or going grocery shopping. It worked most of the time, if it was able to connect. Sometimes it wouldn't connected at all. Only if I force closed AAI and myBMW, deleted cache and then started apps again. Then it worked for my short trips.

Edit: Forgot to mention one thing. Talked with the support about the Connection Issues with BMW Connected 6.5 . It seems like BMW wants to prepare users for which cars will be supported by myBMW. Many older Navs like CIC and NBT (non Evo) will not be supported by myBMW. Therefore, if you have a car with factory CIC or NBT added to your Connected Drive account, it won´t be listet in the "garage" section of BMW Connected 6.5. i for example have an F20 with factory CIC and retrofitted NBT Evo. Since my car had factory CIC, it won´t be supported by myBMW and therefore will not be visible in the garage (even if my smartphone is connected via bluetooth). The Tunneling for AAI won´t work in that case.

**Comment by c-radu** [comment]
What kind of bmw connected you use? 6.4.0 or 6.4.1 ?

**Reply by Tuerkay** [reply]
I'm currently using 6.4.1.6349 but had no problems with 6.4.0 either

**Comment by Kamil2398** [comment]
I think it is worth adding that my bmw app disconnects after about 30 minutes and this is a huge disadvantage of this application.

**Reply by Tuerkay** [reply]
Did you try clearing both apps and reinstall them? I had this issue once after installing myBMW while BMW connected was also installed. Started myBMW saw that my car wasn't shown in garage and removed it again (haven't even connected to car). After that, I had disconnects every 15-20 min. Fixes it by opening BMW connected app settings. Delete data, delete app, restart phone, install myBMW, without starting open app manger, delete data, delete app, restart phone and finally reinstall BMW connected 6.4.1 again. Also cleared AAI cache.

**Reply by Kamil2398** [reply]
Your connection to my BMW is not lost after approximately 30 minutes? Everyone, including Huffman, has confirmed that the problem is with the my bmw app. The BmwConnected app works fine

**Reply by Tuerkay** [reply]
I'm using BMW connected and not myBMW. That's why I wrote you should uninstall myBMW and clean install BMW connected.
